src/providers/osso.js

```diff
@@ -0,0 +1,20 @@
+export default (options) => {
+  return {
+    id: 'osso',
+    name: 'SAML SSO',
+    type: 'oauth',
+    version: '2.0',
+    params: { grant_type: 'authorization_code' },
+    accessTokenUrl: `https://${options.domain}/oauth/token`,
+    authorizationUrl: `https://${options.domain}/oauth/authorize?response_type=code`,
+    profileUrl: `https://${options.domain}/oauth/me`,
+    profile: (profile) => {
+      return {
+        id: profile.id,
+        name: profile.name,
+        email: profile.email
```

Comment:
we currently offer a minimal profile, but could envision adding more attributes as needed. can we `...spread` the profile object? are any of these required?

Reply:
Great question! The docs on this could use improving.

To work with a database, a `id` is required from a profile, it is stored as `accountId`, but name and email are optional as neither are guaranteed from a provider.

From memory:

* If using a database then the `name` and `email` will be used when creating a new user, but if the account exists and is associated with a user, then it will use the `id` to look up the account, then from there look up the `user` associated with that account and return them for the user info when `getSession()` is called.
*  If using JSON Web Tokens without a database, then it doesn't matter what these values are, but if blank they will be blank in the JSON Web Token / Session.
* The full profile returned from a provider is exposed in various callbacks. It is used here for very limited purposes. Returning extra fields here has no impact.